pgjdbc/src/main/java/com/yugabyte/ysql/YBjdbcHostnameVerifier.java

```diff
@@ -0,0 +1,225 @@
+package com.yugabyte.ysql;
+
+import org.postgresql.PGProperty;
+import org.postgresql.core.PGStream;
+import org.postgresql.jdbc.PgConnection;
+import org.postgresql.ssl.PGjdbcHostnameVerifier;
+import org.postgresql.util.GT;
+import org.postgresql.util.HostSpec;
+import org.postgresql.util.PSQLException;
+import org.postgresql.util.PSQLState;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSession;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.security.cert.CertificateParsingException;
+import java.security.cert.X509Certificate;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.postgresql.Driver.*;
+import static org.postgresql.util.internal.Nullness.castNonNull;
+
+public class YBjdbcHostnameVerifier implements HostnameVerifier {
+
+  private static final Logger LOGGER = Logger.getLogger(YBjdbcHostnameVerifier.class.getName());
+  protected static final String GET_SERVERS_QUERY = "select * from yb_servers()";
+  protected Boolean useHostColumn = null;
+
+
+
+
+  private static final int TYPE_DNS_NAME = 2;
+  private static final int TYPE_IP_ADDRESS = 7;
+  private final Properties originalProperties;
+  protected ArrayList<String> currentPublicIps = new ArrayList<>();
+  protected Map<String, String> hostPortMap = new HashMap<>();
+  protected Map<String, String> hostPortMapPublic = new HashMap<>();
+  private final PGStream stream;
+
+  public YBjdbcHostnameVerifier(Properties props, PGStream stream){
+    this.originalProperties = props;
+    this.stream= stream;
+
+  }
+
+  @Override
+  public boolean verify(String hostname, SSLSession session) {
+
+    X509Certificate[] peerCerts;
+    try {
+      peerCerts = (X509Certificate[]) session.getPeerCertificates();
+    } catch (SSLPeerUnverifiedException e) {
+      LOGGER.log(Level.SEVERE,
+          GT.tr("Unable to parse X509Certificate for hostname {0}", hostname), e);
+      return false;
+    }
+    if (peerCerts == null || peerCerts.length == 0) {
+      LOGGER.log(Level.SEVERE,
+          GT.tr("No certificates found for hostname {0}", hostname));
+      return false;
```

Comment:
Should we fail or pass the verification if there are no certs?

Reply:
PgHostnameVerifier fails it if there are no certs available for the hostname